### PR TITLE
Unify Closable pattern & run LeaseManager on a different threadpool

### PR DESCRIPTION
**Pull request by JamesBirdsall**
## Description

The dictionary says it's Closable, not Closeable.

Incorporating feedback from Sreeram:
* ClientEntity and Closable both implemented the same pattern, so unified functionality as ClosableBase.
* Renamed EPH.Pump as EPH.PumpManager, trivial change that makes function much clearer.
* PartitionPump.onReceive/onError short-circuit when closing

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [ ] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [ ] If applicable, the public code is properly documented.
- [ ] Pull request includes test coverage for the included changes.
- [ ] The code builds without any errors.

### CODE REVIEW

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/NamedThreadFactory.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package com.microsoft.azure.eventprocessorhost;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+class NamedThreadFactory implements ThreadFactory {
+	private final String poolName;
+	private int threadCount = 0;
```
Comment (SreeramGarlapati) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/NamedThreadFactory.java]: `volatile int` or `final atomic`

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java [hunk 2]
```diff
@@ -63,6 +65,8 @@
         // Convert all-whitespace prefix to empty string. Convert null prefix to empty string.
         // Then the rest of the code only has one case to worry about.
         this.storageBlobPrefix = (storageBlobPrefix != null) ? storageBlobPrefix.trim() : "";
+        
+        this.mgrThreadpool = Executors.newCachedThreadPool(new NamedThreadFactory("asclm-"));
```
Comment (SreeramGarlapati) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java]: lets expose `this.mgrThreadPool` to be set by users.. as something like - `LeaseTaskExecutor` or a better word!

**Comment by SreeramGarlapati** [comment]
@JamesBirdsall - given that this is taking more iterations - please consider splitting the closable change from that of `EPH` - so that - we can release java client library if we need to (I don't see any urgency to release right now though).

**Comment by JamesBirdsall** [comment]
Need a better way to avoid deadlock and Closable unification is not critical. Not going to do these changes in this release of Java EPH. Avoid the deadlock issue by having this release target Java client 1.0.1 which will not provoke the issue.

**Pull request closed without merging**